src/main/java/core/basesyntax/Quadrangle.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public abstract class Quadrangle extends Figure implements Area {
```

Comment:
Don't use abstract classes to set behavior for classes. Abstract classes and interfaces have different use cases. In this case, since `Quadrangle` is providing an implementation for `area()`, it might be more appropriate to be a concrete class or to move the area calculation to concrete subclasses if the formula differs between them.